File: src/main/java/de/maggiwuerze/xdccloader/util/ProgressWatcherFactory.java
package de.maggiwuerze.xdccloader.util;

import de.maggiwuerze.xdccloader.service.DownloadService;
import lombok.RequiredArgsConstructor;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class ProgressWatcherFactory {

	private final DownloadService downloadService;
	private final ApplicationEventPublisher applicationEventPublisher;

	public FileTransferProgressWatcher getProgressWatcher(Long downloadId) {
		FileTransferProgressWatcher watcher = new FileTransferProgressWatcher(downloadId, applicationEventPublisher, downloadService);
		return watcher;
	}
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/util/ProgressWatcherFactory.kt
package de.maggiwuerze.xdccwebloader.util

import de.maggiwuerze.xdccwebloader.service.DownloadService
import org.springframework.context.ApplicationEventPublisher
import org.springframework.stereotype.Component
import java.util.*

@Component
class ProgressWatcherFactory(
    private val downloadService: DownloadService,
    private val applicationEventPublisher: ApplicationEventPublisher
) {

    fun getProgressWatcher(downloadId: UUID): FileTransferProgressWatcher {
        val watcher: FileTransferProgressWatcher =
            FileTransferProgressWatcher(downloadId, applicationEventPublisher, downloadService)
        return watcher
    }
}
